ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/RecoverySuspendTest.java

```diff
@@ -0,0 +1,205 @@
+        BytemanControlledXAResource.resetCommitCallCounter();
+        BytemanControlledXAResource.resetRollbackCallCounter();
+        BytemanControlledXAResource.resetGreenFlag();
+        BytemanControlledXAResource.setCommitReturn(XAException.XAER_RMFAIL);
+
+        // BytemanControlledRecord.getCommitCallCounter() should be 3 as:
+        // - One invocation from the normal commit procedure should fail
+        // - One invocation from the recovery process should fail
+        // - One invocation from the recovery process should pass
+        startTest(1, 2, 0);
+    }
+
+    private void startTest(int numberOfCommits, int numberOfRollbacks, int timeout) {
+
+        createXATransaction(new BytemanControlledXAResource(), timeout);
+
+        // The Transaction System needs to be disabled, i.e. no new transactions can be created
+        TxControl.disable();
+
+        // Makes sure that the transaction reaper completed all transactions
+        TransactionReaper.transactionReaper().waitForAllTxnsToTerminate();
+
+        // Makes sure that all transactions without a timeout have passed the `prepare` phase
+        waitForTransactionsWithoutTimeout();
+
+        // Synchronization is not needed (i.e. async = true) as isWaitForWorkLeftToDo() == true
+        _manager.suspend(true);
+
+        Assertions.assertEquals(numberOfCommits, BytemanControlledXAResource.getCommitCallCounter(),
+                String.format("BytemanControlledXAResource's getCommitCallCounter is %d but it should have been %d",
+                        BytemanControlledXAResource.getCommitCallCounter(), numberOfCommits));
+
+        Assertions.assertEquals(numberOfRollbacks, BytemanControlledXAResource.getRollbackCallCounter(),
+                String.format("BytemanControlledXAResource's getRollbackCallCounter is %d but it should have been %d",
+                        BytemanControlledXAResource.getRollbackCallCounter(), numberOfRollbacks));
+    }
+
+    private void createXATransaction(XAResource xaResource, int timeout) {
+        TransactionImple tx = new TransactionImple(timeout);
+
+        try {
+            tx.enlistResource(new TestXAResource());
+            tx.enlistResource(xaResource);
+        } catch (SystemException | RollbackException e) {
+            fail();
+        }
+
+        try {
+            tx.commit();
+        } catch (HeuristicRollbackException | SystemException | HeuristicMixedException | RollbackException e) {
+            // it is fine if the transaction fails
+        }
+    }
+
+    private void waitForTransactionsWithoutTimeout() {
+        while (ActionManager.manager().getNumberOfInflightTransactions() > 0) {
+            try {
+                // Let's re-check in 1 second
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private static class DummyXAResourceOrphanFilter implements XAResourceOrphanFilter {
```

Comment:
Why can't we use the real orphan filters?

Reply:
Good point. I'll use `NodeNameXAResourceOrphanFilter`. Thanks

Reply:
Great - thank you